src/main/java/edu/cornell/kfs/sys/aws/batch/AwsSecretValidationStep.java

```diff
@@ -0,0 +1,154 @@
+                    false, AmazonSecretValidationShared.class));
+            
+            AmazonSecretValidationInstance instanceSpaceSecrets = awsSecretService.getPojoFromAwsSecret(AWS_SECRET_NAME_VALIDATION_INSTANCE, 
+                    true, AmazonSecretValidationInstance.class);
+            
+            Map<String, Object> newSecretValues = buildNewSecretValue();
+
+            updateAmazonSecretValidationInstance(instanceSpaceSecrets, newSecretValues);
+
+            awsSecretService.updatePojo(AWS_SECRET_NAME_VALIDATION_INSTANCE, true, instanceSpaceSecrets);
+
+            confirmInstanceSecretValues(newSecretValues, awsSecretService.getPojoFromAwsSecret(AWS_SECRET_NAME_VALIDATION_INSTANCE, 
+                    true, AmazonSecretValidationInstance.class));
+
+            LOG.info("execute, The values were retrieved from cache as expected");
+            awsSecretService.logCacheStatus();
+            awsSecretService.clearCache();
+
+            AmazonSecretValidationInstance amazonInstanceSpaceSecretsAfterProcessing = awsSecretService.getPojoFromAwsSecret(
+                    AWS_SECRET_NAME_VALIDATION_INSTANCE, true, AmazonSecretValidationInstance.class);
+            confirmInstanceSecretValues(newSecretValues, amazonInstanceSpaceSecretsAfterProcessing);
+
+            LOG.info("excute, The values were retrieved from Amazon Secret Manager as expected");
+            awsSecretService.logCacheStatus();
+
+        } catch (JsonMappingException e) {
+            LOG.error("execute, had an error mapping AWS Secrets to POJO", e);
+            throw new RuntimeException(e);
+        } catch (JsonProcessingException e) {
+            LOG.error("execute, had an error process AWS Secrets", e);
+            throw new RuntimeException(e);
+        }
+        
+        validateUsingNewAWSServiceFromSpring();
+
+        awsSecretService.clearCache();
+        LOG.info("execute, ending job");
+        return true;
+    }
+    
+    private void confirmAmazonSecretValidationShared(AmazonSecretValidationShared sharedSpaceSecrets) {
+        String errorMessage = StringUtils.EMPTY;
+        if (!StringUtils.equals(sharedSpaceSecrets.getLogin_password(), EXPECTED_SHARED_PASSWORD)) {
+            errorMessage = errorMessage + "The expected passwword was not found.";
+        }
+        if (!StringUtils.equals(sharedSpaceSecrets.getLogin_username(), EXPECTED_SHARED_USER_NAME)) {
+            errorMessage = errorMessage + " The expected user name was not found.";
+        }
+
+        if (StringUtils.isNotBlank(errorMessage)) {
+            throw new RuntimeException(errorMessage);
+        } else {
+            LOG.info("confirmAmazonSecretValidationShared, the shared secrets contained the expected values");
+        }
+    }
+
+    private Map<String, Object> buildNewSecretValue() {
+        Map<String, Object> values = new HashMap<String, Object>();
```

Comment:
The part to the right of the equals sign could be simplified as "new HashMap<>()" if needed.

Reply:
updated